### ISSUE TITLE
Dragging Pie Slices

### ISSUE THREAD

**Issue by jgathairu** [amcharts/amcharts4#2977]
Have picked up the example of dragging pie from https://www.amcharts.com/demos/dragging-pie-slices/#code , works perfect but when I put custom data through datasource.url am experiencing the following type of chart
![Error](https://user-images.githubusercontent.com/29246216/94987777-a3a6e800-0571-11eb-997a-b1065a06b99d.PNG)

On the console the error is as follows
![Error2](https://user-images.githubusercontent.com/29246216/94987816-cfc26900-0571-11eb-8a87-48899743e47a.PNG) dummyDataItem shows undefined. On the test example provided on the websites the dummyDataItem has items.

My datasource URL json data is as shown
[SampleJson.txt](https://github.com/amcharts/amcharts4/files/5321352/SampleJson.txt)
. Any insights as to why my code could be having issues is welcome. Have attached the full source code (javascript)
[SampleJavascript.txt](https://github.com/amcharts/amcharts4/files/5321344/SampleJavascript.txt)

**Comment by martynasma** [comment]
This question was reviewed by amCharts staff and was determined neither a) bug nor b) simple question that can be answered without doing custom development or reviewing user code.

I'm sorry, but we won't be able to answer your question here.

If you have an amCharts support subscription, we suggest contacting our helpdesk. If you don't, you might consider acquiring one. For more details on procuring support please refer to [this page](https://www.amcharts.com/support/support-info/get-support/).

**Comment by jgathairu** [comment]
Thank you for the reply @amcharts. Am a licensed user but fortunately, after a few hours on it, I managed to resolve the issue.  The error occur because the call back to the datasource URL had not completed by the time the code it hit. To avoid this, I wrapped the chart under an ajax only calling it after It received a promise, that way the code execution for the chart starts to run when data has been reveived from the URL. Issue closed
